datafusion/expr/src/expr.rs

```diff
@@ -2069,6 +2069,11 @@ impl Expr {
             _ => None,
         }
     }
+
+    /// Check if the Expr is literal
```

Comment:
As a minor nit -- I wonder if you changed this to `as_literal` the code below could get simpler

```rust
pub fn as_literal(&self) -> Option<&Literal> {
 if let Expr::Literal(lit, _) = self { Some(lit) } else { None } 
}
```

Then some of the matching might be simpler below

Reply:
Yeah, neat!